modal/image.py

```diff
@@ -1299,6 +1299,140 @@ def build_dockerfile(version: ImageBuilderVersion) -> DockerfileSpec:
             gpu_config=parse_gpu_config(gpu),
         )
 
+    def uv_sync(
+        self,
+        uv_project_dir: str = "./",  # Path to local uv mananged project
+        *,
+        force_build: bool = False,  # Ignore cached builds, similar to 'docker build --no-cache'
+        group: Optional[str] = None,  # Dependency group to install using `uv sync --group`
+        optional: Optional[str] = None,  # Optional dependencies to install using `uv sync --optional`
```

Comment:
AFAICT there's no `--optional` option for `uv sync` ... I see one for `uv add` and that places the added dependency in the `optional.dependency` table but not super clear to me how to request this at the `uv sync` time.

Reply:
Yea, it's `extra`. Thank you for catching this!